mitodl/mitxpro
# Streamlined Wagtail configuration and seed data provisioning

#### Pre-Flight checklist


#### What are the relevant tickets?
No ticket

#### What's this PR do?
Streamlines the steps to get this site actually functional. 

- All the necessary Wagtail changes can be applied by running a single management command
- The seed data script automatically ensures that those Wagtail changes are made each time it is executed
- The Wagtail changes are idempotent

#### How should this be manually tested?
1. Create a completely fresh database and name it `postgres` (which is the name of the db that the xpro app uses)
    - **NOTE:** If you'd like a quick script to back up your database, set a completely new database as your xpro app database, and restore the database that you backed up, let me know.
2. Bring up the docker containers
3. Run the seed data management command: `manage.py seed_data`
4. Visit `/catalog/`

You should see the expected seeded courses and programs. 

#### Any background context you want to provide?
This should have been done a long time ago. I had been frustrated enough times by this highly manual process that I felt the time was right to gather up my various helper scripts and turn it into something that everyone can use.

#### Screenshots (if appropriate)
![ss 2020-03-09 at 15 28 56 ](https://user-images.githubusercontent.com/14932219/76251551-8ef1b400-621d-11ea-9d7f-4f75d1dbcfb0.png)
